src/OrchardCore.Modules/OrchardCore.Seo/Indexes/SeoMetaPartIndex.cs

```diff
@@ -0,0 +1,32 @@
+﻿using OrchardCore.ContentManagement;
+using OrchardCore.Seo.Models;
+using YesSql.Indexes;
+
+namespace OrchardCore.Seo.Indexes;
+
+public class SeoMetaPartIndex : MapIndex
+{
+    public string PageTitle { get; set; }
+}
+
+public class SeoMetaPartIndexProvider : IndexProvider<ContentItem>
+{
+    public override void Describe(DescribeContext<ContentItem> context)
+    {
+        context.For<SeoMetaPartIndex>()
+            .When(contentItem => contentItem.Has<SeoMetaPart>())
+            .Map(contentItem =>
+            {
+                var containedPart = contentItem.As<SeoMetaPart>();
```

Comment:
You can validate here if ContentItem is Published AND Latest to index only records when the content item is actually producing a working URL.